spec/prog/postgres/postgres_server_nexus_spec.rb

```diff
@@ -852,16 +852,32 @@
       expect(postgres_server).to receive(:resource).and_return(postgres_resource)
       expect(postgres_server).to receive(:trigger_failover).and_return(false)
       expect(nx).to receive(:available?).and_return(true)
+      expect(nx).to receive(:decr_recycle)
       expect { nx.unavailable }.to hop("wait")
     end
 
     it "buds restart if the server is not available" do
-      postgres_resource = instance_double(PostgresResource, ongoing_failover?: false)
-      expect(postgres_server).to receive(:resource).and_return(postgres_resource)
+      postgres_resource = instance_double(PostgresResource, ongoing_failover?: false, id: "resource-id")
```

Comment:
We should avoid mocking for this. Setup the database state, run the label, and then check that the expected database changes were made.